### PR TITLE
IR-471:registry operator: Add missing env var, wait for rbac

### PR DESCRIPTION
**Pull request by csrwng**
**What this PR does / why we need it**:
Add the missing OPERATOR_IMAGE_VERSION required by the new addition of featuregate in the Image-registry operator

In some cases the image registry operator starts (and crashes) before
the CVO has given its service account proper rbac. This commit adds a
wait for access to the infrastructure resource before allowing the
operator to start.

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #OCPBUGS-34678

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [x] Relevant issues have been referenced.

### CODE REVIEW

File: control-plane-operator/controllers/hostedcontrolplane/registryoperator/reconcile.go [hunk 1]
```diff
@@ -46,6 +46,15 @@ echo "{{ .WorkerNamespace }}" > "{{ .TokenDir }}/namespace"
 cp "{{ .CABundle }}" "{{ .TokenDir }}/ca.crt"
 export KUBERNETES_SERVICE_HOST=kube-apiserver
 export KUBERNETES_SERVICE_PORT=$KUBE_APISERVER_SERVICE_PORT
+
+while true; do
```
[the remark this answers is not in the document]
Comment (stevekuznetsov) [reply control-plane-operator/controllers/hostedcontrolplane/registryoperator/reconcile.go:50]: The `break` on 52 - or do you mean a timeout?

**Review by stevekuznetsov — APPROVED** [review]

File: control-plane-operator/controllers/hostedcontrolplane/registryoperator/reconcile.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (bryan-cox) [reply control-plane-operator/controllers/hostedcontrolplane/registryoperator/reconcile.go:50]: Yeah a timeout

[the remark this answers is not in the document]
Comment (csrwng) [reply control-plane-operator/controllers/hostedcontrolplane/registryoperator/reconcile.go:50]: I don't think so... if we don't have the right RBAC, then it's going to break if it starts anyway

**Comment by deepsm007** [comment]
/retitle IR-471:registry operator: Add missing env var, wait for rbac

**Pull request merged**